Hi @asluk ! I haven't debugged yet to see what's going wrong, but I wanted to give you a heads up that I haven't had a successful test run ~

```
====================================================================== 
ERROR: test_URLEncodedIdentifiers (__main__.TestUsdStage)
 ---------------------------------------------------------------------- 
Traceback (most recent call last): 

File "/scratch/build_local/dev/src/inst/fedora-gcc64-opt/pxr/tests/testUsdStage", 
line 33, in test_URLEncodedIdentifiers
  stage = Usd.Stage.Open("Libeccio%20LowFBX.usda") 
pixar.Tf.ErrorException:  Error in 'UsdStage::Open' at line 879 in file pxr/usd/usd/stage.cpp : 
   'Failed to open layer @Libeccio%20LowFBX.usda@'
```

--- a/pxr/usd/sdf/assetPathResolver.cpp
+++ b/pxr/usd/sdf/assetPathResolver.cpp
@@ -258,6 +258,12 @@ Sdf_GetAnonLayerIdentifierTemplate(
     const string& tag)
 {
     string idTag = tag.empty() ? tag : TfStringTrim(tag);
+
+    // Ensure that URL-encoded characters are not misinterpreted as
+    // format strings to TfStringPrintf in Sdf_ComputeAnonLayerIdentifier.
+    // See discussion in https://github.com/PixarAnimationStudios/USD/pull/2022
+    idTag = TfStringReplace(idTag, "%", "%%");
+
     return _Tokens->AnonLayerPrefix.GetString() + "%p" +
         (idTag.empty() ? idTag : ":" + idTag);
 }

--- a/pxr/usd/usd/CMakeLists.txt
+++ b/pxr/usd/usd/CMakeLists.txt
@@ -458,6 +458,11 @@ pxr_install_test_dir(
 )
 
 pxr_install_test_dir(
+    SRC testenv/testUsdStage
+    DEST testUsdStage
+)
+
+pxr_install_test_dir(
     SRC testenv/testUsdAppliedAPISchemas
     DEST testUsdAppliedAPISchemas_AutoApplyDisabled
 )

--- a/pxr/usd/usd/testenv/testUsdStage.py
+++ b/pxr/usd/usd/testenv/testUsdStage.py
@@ -29,6 +29,10 @@ from pxr import Sdf,Usd,Tf
 allFormats = ['usd' + c for c in 'ac']
 
 class TestUsdStage(unittest.TestCase):
+    def test_URLEncodedIdentifiers(self):
+        stage = Usd.Stage.Open("Libeccio%20LowFBX.usda")
+        assert stage
+
     def test_Repr(self):
         stage = Usd.Stage.CreateInMemory()
 

--- a/pxr/usd/usd/testenv/testUsdStage/Libeccio%20LowFBX.usda
+++ b/pxr/usd/usd/testenv/testUsdStage/Libeccio%20LowFBX.usda
@@ -1,0 +1,17 @@
+#usda 1.0
+(
+    defaultPrim = "hello"
+)
+
+def Xform "hello"
+{
+    custom double3 xformOp:translate = (4, 5, 6)
+    uniform token[] xformOpOrder = ["xformOp:translate"]
+
+    def Sphere "world"
+    {
+        float3[] extent = [(-2, -2, -2), (2, 2, 2)]
+        color3f[] primvars:displayColor = [(0, 0, 1)]
+        double radius = 2
+    }
+}

